src/main/java/org/jabref/gui/maintable/MainTableTooltip.java

```diff
@@ -18,28 +17,24 @@ public class MainTableTooltip extends Tooltip {
 
     private final PreviewViewer preview;
     private final PreferencesService preferences;
+    private final VBox tooltipContent = new VBox();
+    private final Label fieldValueLabel = new Label();
 
-    public MainTableTooltip(BibDatabaseContext databaseContext, DialogService dialogService, PreferencesService preferences,
-                            StateManager stateManager, ThemeManager themeManager, TaskExecutor taskExecutor) {
-
+    public MainTableTooltip(BibDatabaseContext databaseContext, DialogService dialogService, PreferencesService preferences, ThemeManager themeManager, TaskExecutor taskExecutor) {
         this.preferences = preferences;
-        this.preview = new PreviewViewer(databaseContext, dialogService, preferences, stateManager, themeManager, taskExecutor);
+        this.preview = new PreviewViewer(databaseContext, dialogService, preferences, themeManager, taskExecutor);
+        this.setShowDelay(Duration.seconds(1));
+        this.tooltipContent.getChildren().addAll(fieldValueLabel, preview);
     }
 
     public Tooltip createTooltip(BibEntry entry, String fieldValue) {
-        preview.setLayout(preferences.getPreviewPreferences().getSelectedPreviewLayout());
-        preview.setEntry(entry);
-        this.setShowDelay(Duration.seconds(1));
-        Label label = new Label(fieldValue);
-
-        if (preferences.getPreviewPreferences().shouldShowPreviewEntryTableTooltip()) {
-            VBox vBox = new VBox();
-            label.setText(label.getText() + "\n ");
-            vBox.getChildren().add(label);
-            vBox.getChildren().add(preview);
-           this.setGraphic(vBox);
+        fieldValueLabel.setText(fieldValue + "\n");
```

Comment:
This can be moved to the if block

Please swap the conditions - typically, one should have the positive case first. Alternatively, "fail fast". Keep the negative thing, but then do "return this;" and then without `else` do the rest.

The indent of line 37 "this.setGraphic" is off

Reply:
In both cases it should be updated, as it is also used in the `tooltipContent` line 27.

Reply:
Oh, wow, something which is not directly seeable while browsing through the github diff 😅